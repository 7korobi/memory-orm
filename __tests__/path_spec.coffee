require "./models/index"
Mem = require "../src/index"

describe "path", ->
  test "children snapshot", ->
    expect Mem.Query.paths.pluck("id", "paths.hash")
    .toMatchSnapshot()
  test "parent snapshot", ->
    expect Mem.Query.paths.pluck("id", "path")
    .toMatchSnapshot()
  undefined


