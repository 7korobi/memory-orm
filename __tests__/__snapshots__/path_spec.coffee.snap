// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`path children snapshot 1`] = `
list [
  Array [
    "book-1-aaa",
    Object {},
  ],
  Array [
    "book-1-bbb",
    Object {
      "book-1-bbb-ccc": model {
        "_id": "book-1-bbb-ccc",
        "book_id": "book-1",
        "folder_id": "book",
        "idx": "ccc",
        "path_id": "book-1-bbb",
      },
      "book-1-bbb-ddd": model {
        "_id": "book-1-bbb-ddd",
        "book_id": "book-1",
        "folder_id": "book",
        "idx": "ddd",
        "path_id": "book-1-bbb",
      },
    },
  ],
  Array [
    "book-1-bbb-ccc",
    Object {},
  ],
  Array [
    "book-1-bbb-ddd",
    Object {
      "book-1-bbb-ddd-eee": model {
        "_id": "book-1-bbb-ddd-eee",
        "book_id": "book-1",
        "folder_id": "book",
        "idx": "eee",
        "path_id": "book-1-bbb-ddd",
      },
    },
  ],
  Array [
    "book-1-bbb-ddd-eee",
    Object {},
  ],
  Array [
    "book-1-ccc-ddd",
    Object {},
  ],
  Array [
    "book-1-ccc-eee",
    Object {},
  ],
  Array [
    "book-1-ddd-eee",
    Object {},
  ],
  Array [
    "book-1-fff-ggg-hhh-iii",
    Object {},
  ],
]
`;

exports[`path parent snapshot 1`] = `
list [
  Array [
    "book-1-aaa",
    null,
  ],
  Array [
    "book-1-bbb",
    null,
  ],
  Array [
    "book-1-bbb-ccc",
    model {
      "_id": "book-1-bbb",
      "book_id": "book-1",
      "folder_id": "book",
      "idx": "bbb",
    },
  ],
  Array [
    "book-1-bbb-ddd",
    model {
      "_id": "book-1-bbb",
      "book_id": "book-1",
      "folder_id": "book",
      "idx": "bbb",
    },
  ],
  Array [
    "book-1-bbb-ddd-eee",
    model {
      "_id": "book-1-bbb-ddd",
      "book_id": "book-1",
      "folder_id": "book",
      "idx": "ddd",
      "path_id": "book-1-bbb",
    },
  ],
  Array [
    "book-1-ccc-ddd",
    null,
  ],
  Array [
    "book-1-ccc-eee",
    null,
  ],
  Array [
    "book-1-ddd-eee",
    null,
  ],
  Array [
    "book-1-fff-ggg-hhh-iii",
    null,
  ],
]
`;
